crypto/src/tests/bls12381_tests.rs

```diff
@@ -400,3 +403,75 @@ async fn signature_service() {
     // Verify the signature we received.
     assert!(pk.verify(digest.as_ref(), &signature).is_ok());
 }
+
+// Checks if the private keys zeroed out
+#[test]
+fn test_zeroization_priv_key() {
+    let kp = keys().pop().unwrap();
+    let mut sk = kp.private();
+    let mut sk_bytes = Vec::new();
+    sk_bytes.extend_from_slice(sk.as_ref());
+
+    let ptr = std::ptr::addr_of!(sk.privkey) as *const u8;
+    let bytes_ptr = &sk.as_ref()[0] as *const u8;
+
+    unsafe {
+        let mut vec = Vec::new();
+        for i in 0..BLS12381PrivateKey::LENGTH {
+            vec.push(*bytes_ptr.add(i));
+        }
+        assert_eq!(&vec[..], &sk_bytes[..]);
+    }
+
+    sk.zeroize();
+
+    // Check that self.privkey is zeroized
+    unsafe {
+        for i in 0..BLS12381PrivateKey::LENGTH {
+            assert!(*ptr.add(i) == 0);
+        }
+    }
+
+    // Check that self.bytes is zeroized
+    unsafe {
+        let mut vec = Vec::new();
+        for i in 0..BLS12381PrivateKey::LENGTH {
+            vec.push(*bytes_ptr.add(i));
+        }
+        assert_ne!(&vec[..], &sk_bytes[..]);
+    }
+}
+
+// Checks if the private keys zeroed out
+#[test]
+fn test_zeroization_on_drop() {
+    let ptr: *const u8;
+    let bytes_ptr: *const u8;
+
+    let mut sk_bytes = Vec::new();
+
+    {
+        let kp = keys().pop().unwrap();
+        let sk = kp.private();
+        sk_bytes.extend_from_slice(sk.as_ref());
+
+        ptr = std::ptr::addr_of!(sk.privkey) as *const u8;
+        bytes_ptr = &sk.as_ref()[0] as *const u8;
+    }
```

Comment:
I for one am OK testing just zeroize on drop, and inferring that if it works on drop, it works everywhere.
But I won't fight against the inclusion of one more test, of course.

Reply:
I think that makes sense, drop calls zeroize itself so it makes no sense to test manually calling zeroize.

Reply:
I've changed the PR to only zeroize on drop